app/src/main/java/ch/epfl/sdp/blindly/main_screen/match/MatchPageFragment.kt

```diff
@@ -117,11 +122,20 @@ class MatchPageFragment : Fragment(), CardStackListener {
     }
 
     /**
-     * Do some actions when the card is swiped
+     * When the card is swiped right, add the uid to the liked profiles
      *
      * @param direction the direction the card is swiped (left, right)
      */
+    @RequiresApi(Build.VERSION_CODES.N)
     override fun onCardSwiped(direction: Direction) {
+        if (direction == Direction.Right) {
+            likedUserId = currentCardUid
+            val updatedLikesList = currentUser.likes?.plus(likedUserId)
+            viewLifecycleOwner.lifecycleScope.launch {
+                userRepository.updateProfile(currentUserId, "likes", updatedLikesList)
```

Comment:
Maybe use the const in the User class instead: LIKES

Reply:
Thank you, I forgot about that !